src/parcsr_ls/par_mgr.c

```diff
@@ -3487,43 +3465,74 @@ hypre_MGRSetGlobalSmoothType( void *mgr_vdata, HYPRE_Int gsmooth_type )
    return hypre_error_flag;
 }
 
-/* Set global smoothing type for mgr solver */
+/*--------------------------------------------------------------------------
+ * hypre_MGRSetLevelSmoothType
+ *
+ * Set global smoothing type at each MGR level.
+ *--------------------------------------------------------------------------*/
+
 HYPRE_Int
-hypre_MGRSetLevelSmoothType( void *mgr_vdata, HYPRE_Int *gsmooth_type )
+hypre_MGRSetLevelSmoothType( void       *mgr_vdata,
+                             HYPRE_Int  *gsmooth_type )
 {
    hypre_ParMGRData   *mgr_data = (hypre_ParMGRData*) mgr_vdata;
-   HYPRE_Int i;
-   HYPRE_Int max_num_coarse_levels = (mgr_data -> max_num_coarse_levels);
-   hypre_TFree((mgr_data -> level_smooth_type), HYPRE_MEMORY_HOST);
+   HYPRE_Int           max_num_coarse_levels = (mgr_data -> max_num_coarse_levels);
+   HYPRE_Int          *level_smooth_type, i;
+   char                msg[1024];
 
-   HYPRE_Int *level_smooth_type = hypre_CTAlloc(HYPRE_Int, max_num_coarse_levels, HYPRE_MEMORY_HOST);
+   /* Set level_smooth_type array */
+   level_smooth_type = hypre_CTAlloc(HYPRE_Int, max_num_coarse_levels, HYPRE_MEMORY_HOST);
+   hypre_TFree((mgr_data -> level_smooth_type), HYPRE_MEMORY_HOST);
    if (gsmooth_type != NULL)
    {
       for (i = 0; i < max_num_coarse_levels; i++)
       {
-         level_smooth_type[i] = gsmooth_type[i];
+         /* For meaningful values of global smoothing type, the option set via
```

Comment:
Should this check be made once in mgr_setup, since we would know then whether level_smoother and level_smooth_type are both set then set precedence and invalidate the level_smooth_type entry as appropriate? Perhaps before this line: https://github.com/hypre-space/hypre/blob/4f5d6ad50639d6ba8f1250d569221e1f13ebbe1e/src/parcsr_ls/par_mgr_setup.c#L958